src/candy-machine/instructions/initialize-candy-machine.ts

```diff
@@ -22,8 +26,210 @@ import { TokenStandard } from '@metaplex-foundation/mpl-token-metadata';
 import {
   AUTH_RULES,
   AUTH_RULES_ID,
+  BOT_TAX,
+  D_PUBLISHER_SYMBOL,
+  MIN_COMPUTE_PRICE,
   MIN_COMPUTE_PRICE_IX,
 } from '../../constants';
+import {
+  Umi,
+  PublicKey as UmiPublicKey,
+  percentAmount,
+  some,
+  lamports,
+  publicKey,
+  generateSigner,
+} from '@metaplex-foundation/umi';
+import {
+  create,
+  Creator as CoreCmCreator,
+  createLutForCandyMachine,
+} from '@metaplex-foundation/mpl-core-candy-machine';
+import { toUmiGroups } from '../../utils/core-candy-machine';
+import { ComicIssueCMInput } from '../../comic-issue/dto/types';
+import { GuardParams } from '../dto/types';
+import {
+  JsonMetadataCreators,
+  toLegacyGroups,
+} from '../../utils/candy-machine';
+import { initializeRecordAuthority } from './initialize-record-authority';
+import { solFromLamports } from '../../utils/helpers';
+import { setComputeUnitPrice } from '@metaplex-foundation/mpl-toolbox';
+import { createLookupTable } from '../../utils/lookup-table';
+import {
+  ASSOCIATED_TOKEN_PROGRAM_ID,
+  TOKEN_PROGRAM_ID,
+} from '@solana/spl-token';
+
+// Core CandyMachine
+export async function createCoreCandyMachine(
+  umi: Umi,
+  collectionNftAddress: UmiPublicKey,
+  comicIssue: ComicIssueCMInput,
+  royaltyWallets: JsonMetadataCreators,
+  guardParams: GuardParams,
+  isPublic?: boolean,
+) {
+  const candyMachineKey = generateSigner(umi);
+  const creators: CoreCmCreator[] = royaltyWallets.map((item) => {
+    return {
+      address: publicKey(item.address),
+      percentageShare: item.share,
+      verified: false,
+    };
+  });
+
+  const createCmBuilder = await create(umi, {
+    candyMachine: candyMachineKey,
+    collection: collectionNftAddress,
+    collectionUpdateAuthority: umi.identity,
+    sellerFeeBasisPoints: percentAmount(
+      //TODO: Check this if it's giving correct percentage
+      Number((comicIssue.sellerFeeBasisPoints / 100).toFixed(2)),
+      2,
+    ),
+    itemsAvailable: guardParams.supply,
+    guards: {
+      botTax: some({
+        lamports: lamports(BOT_TAX),
+        lastInstruction: true,
+      }),
+    },
+    groups: toUmiGroups(umi, guardParams, isPublic),
+    creators: [
+      {
+        address: umi.identity.publicKey,
+        verified: true,
+        percentageShare: 0,
+      },
+      ...creators,
+    ],
+    configLineSettings: some({
+      prefixName: '',
+      nameLength: 32,
+      prefixUri: '',
+      uriLength: 200,
+      isSequential: false,
+    }),
+  });
+
+  const recentSlot = await umi.rpc.getSlot({ commitment: 'confirmed' });
+  // TODO: check if it requires more compute and why it fails with (Program fails to compelete) without skipPreflight
+  const builder = setComputeUnitPrice(umi, {
+    microLamports: MIN_COMPUTE_PRICE,
+  }).add(createCmBuilder);
+  await builder.sendAndConfirm(umi, {
+    send: { commitment: 'confirmed', skipPreflight: true },
+  });
+
+  let lookupTable: UmiPublicKey;
```

Comment:
should we remove `let` variable here and instead make a return inside of `try` block? `catch` should handle/throw some error I guess?

Reply:
So i did this specifically , sometimes tx fails so in cas lookup table failed to be created we still get candymachine that was created as lookup table is optional and we can create it with another tx

Reply:
Gotcha, but still we can make it as a const and inside try block. If in either case we need to return candy machine, we can do it from catch block too. I'll leave it on you, whatever you prefer